kernel/aster-nix/src/device/mod.rs

```diff
@@ -47,3 +47,20 @@ pub fn init() -> Result<()> {
     pty::init()?;
     Ok(())
 }
+
+pub fn get_device(dev: usize) -> Result<Arc<dyn Device>> {
```

Comment:
Why not use `DeviceId`  here?
https://github.com/asterinas/asterinas/blob/9e4775d9e2c013899ce50dbc33250dbd5084a5a4/kernel/aster-nix/src/fs/device.rs#L46

Reply:
Thank you for pointing this out, but the creation of `DeviceId` requires `major ID` and `minor ID`, and the parameter device number `dev` of `mknod` is composed of `major ID` and `minor ID`. Therefore, what we need is to convert the `dev` parameter into `major ID` and `minor ID` respectively, and `DeviceId` does not seem to provide this function , because it first requires `major ID` and `minor ID` to create it, but if we can calculate `major ID` and `minor ID`, why do we need to create a `DeviceId`?

```rust
impl DeviceId {
    pub fn new(major: u32, minor: u32) -> Self {
        let major = major as u64;
        let minor = minor as u64;
        Self(
            (major & 0xffff_f000) << 32
                | (major & 0x0000_0fff) << 8
                | (minor & 0xffff_ff00) << 12
                | (minor & 0x0000_00ff),
        )
    }
    ...
}
```

Reply:
In general, it is a good idea for the kernel to use the more powerful and safe `DeviceId`. This conversion from its raw value to `DeviceId` is zero cost. So using `DeviceId` is always more preferable.

```rust
impl From<u64> from DeviceId {
    fn from(raw: u64) {
        Self(raw)
    }
}
```

Reply:
Done